src/spikeinterface/core/zarrextractors.py

```diff
@@ -315,6 +316,12 @@ def add_properties_and_annotations(
     prop_group = zarr_group.create_group("properties")
     for key in recording_or_sorting.get_property_keys():
         values = recording_or_sorting.get_property(key)
+        if values.dtype.kind == "O":
+            try:
+                values = values.astype("str")
+            except Exception as e:
```

Comment:
I would encode objects with msgpack.

https://zarr.readthedocs.io/en/stable/tutorial.html#string-arrays

Of the protocols of zarr is the one that is easily available on other langugaes.

Reply:
Or json if you prefer, probably the performance of msgpack does not mater in this case.

Reply:
I think we should not encode all objects. Say you have a `Sorting` object as a property. I think we should skip saving those to zarr. Ok to use the `msgpack` codec, but I'm not sure why this simple fix is working then, since the error came from the `KSLabel`, which can be `good`, `mua` or `noise` (so it's a variable string). Probably, using `astype(str)` converts it to an `S` type, with is handled differently than `U` dtype. Any thoughts?

Reply:
What was the error you were getting?

I think that zarr in python should handle the old fix width unicode just fine.

Reply:
can't remember the exact error, but it was about missing an `object_codec`

Reply:
If this is solving ithe problem your original object was neither byte string or unicode:

https://numpy.org/doc/stable/reference/generated/numpy.dtype.kind.html

I would check why KSLabel is not transformed to string when you load it but this code here would solve your problem. Want to merge? I would go and put the specific exception that you are getting in your code though.

Reply:
No rush! I think that string arrays are automatically loaded as Objects by `pandas`, and KS output is loaded with pandas from the generated csv files. I'll post the error I was getting tomorrow.

Reply:
Dude, this shit works:

```python
import numpy as np



from spikeinterface.core import generate_recording


recording1 = generate_recording()
recording2 = generate_recording()

array = np.array([recording1, recording2]) 

array.astype("str")
```

No exception thrown : /

Reply:
Ouch! Maybe because of the `__repr__` function?

Reply:
Yes, I was just reading the spec. Apparently they use `__str__` by default. So it is difficult to make this fail. Another example:

```python


recording1 = generate_recording()
recording2 = generate_recording()
import random

class RandomAttributes:
    def __init__(self):
        self.attribute1 = random.randint(1, 10)
        self.attribute2 = random.choice(['red', 'blue', 'green'])
        self.attribute3 = random.uniform(0.0, 1.0)

# Create an instance of the class
obj1 = RandomAttributes()
obj2 = RandomAttributes()   

array = np.array([obj1, obj2]) 

new_array =array.astype("str")
new_array
```

 I was hoping we could use casting safe rules but unfortunatley this fails:
```python 
original_string_array = np.array(["this", "that", ])
original_string_array.astype("object").astype("str", casting="safe")
 ```

Reply:
Yeah, I think the quicker error would be to indicate in the respective `pd.load_csv` that the type of that column is string.

Reply:
Good idea and probably a better fix!

Reply:
Done, should be better now. Let me check if that solves the issue I was having :)

Reply:
Great. Approved.